From e9bca058d6927f4998d26eb0ef68ac26d4d446c5 Mon Sep 17 00:00:00 2001
From: Sam Ravnborg <srn@skov.dk>
Date: Wed, 15 Feb 2017 23:39:40 +0100
Subject: [PATCH] sync files with 2012.10 U-Boot

Copy files to kill random whitespace changes etc.
Keep diff smaller
---
 common/cmd_nand.c     | 11 ---------
 common/env_flash.c    | 66 ++++++++++++++++++++++++++++-----------------------
 include/dataflash.h   |  1 -
 include/environment.h | 12 ++++------
 include/flash.h       |  3 ---
 tools/Makefile        |  1 +
 6 files changed, 42 insertions(+), 52 deletions(-)

diff --git a/common/cmd_nand.c b/common/cmd_nand.c
index 9a5fe6f..e24ed7f 100644
--- a/common/cmd_nand.c
+++ b/common/cmd_nand.c
@@ -579,11 +579,6 @@ int do_nand(cmd_tbl_t * cmdtp, int flag, int argc, char * const argv[])
 			else if (getc() == 'y') {
 				puts("y");
 				if (getc() == '\r')
-					/**
-					 * @author Martin Bjærre <mab@rosetechnology.dk>
-					 * This functionality is removed as it endangers bad block information
-					 *
-					 */
 					opts.scrub = 1;
 				else {
 					puts("scrub aborted\n");
@@ -692,12 +687,6 @@ int do_nand(cmd_tbl_t * cmdtp, int flag, int argc, char * const argv[])
 			if (read)
 				ret = nand->read_oob(nand, off, &ops);
 			else
-				/**
-                                 * @author Martin Bjærre <mab@rosetechnology.dk>
-                                 *
-                                 * This functionality is removed as it endangers bad block information
-                                 *
-                                 */
 				ret = nand->write_oob(nand, off, &ops);
 		} else if (raw) {
 			ret = raw_access(nand, addr, off, pagecount, read);
diff --git a/common/env_flash.c b/common/env_flash.c
index 08a6ae6..aa970d4 100644
--- a/common/env_flash.c
+++ b/common/env_flash.c
@@ -24,7 +24,7 @@
  * MA 02111-1307 USA
  */
 
-//#define DEBUG
+/* #define DEBUG */
 
 #include <common.h>
 #include <command.h>
@@ -40,12 +40,12 @@ DECLARE_GLOBAL_DATA_PTR;
 #define CMD_SAVEENV
 #elif defined(CONFIG_ENV_ADDR_REDUND)
 #error CONFIG_ENV_ADDR_REDUND must have CONFIG_CMD_SAVEENV & CONFIG_CMD_FLASH
-#endif /* CONFIG_CMD_SAVEENV || CONFIG_CMD_FLASH */
+#endif
 
 #if defined(CONFIG_ENV_SIZE_REDUND) &&	\
 	(CONFIG_ENV_SIZE_REDUND < CONFIG_ENV_SIZE)
 #error CONFIG_ENV_SIZE_REDUND should not be less then CONFIG_ENV_SIZE
-#endif /* CONFIG_ENV_SIZE_REDUND */
+#endif
 
 char *env_name_spec = "Flash";
 
@@ -53,17 +53,17 @@ char *env_name_spec = "Flash";
 env_t *env_ptr = &environment;
 
 static env_t *flash_addr = (env_t *)CONFIG_ENV_ADDR;
+
 #else /* ! ENV_IS_EMBEDDED */
 
 env_t *env_ptr = (env_t *)CONFIG_ENV_ADDR;
-
 static env_t *flash_addr = (env_t *)CONFIG_ENV_ADDR;
 #endif /* ENV_IS_EMBEDDED */
 
 #if defined(CMD_SAVEENV) || defined(CONFIG_ENV_ADDR_REDUND)
 /* CONFIG_ENV_ADDR is supposed to be on sector boundary */
 static ulong end_addr = CONFIG_ENV_ADDR + CONFIG_ENV_SECT_SIZE - 1;
-#endif /* CMD_SAVEENV || CONFIG_ENV_ADDR_REDUND */
+#endif
 
 #ifdef CONFIG_ENV_ADDR_REDUND
 static env_t *flash_addr_new = (env_t *)CONFIG_ENV_ADDR_REDUND;
@@ -88,6 +88,7 @@ int env_init(void)
 	crc1_ok = crc32(0, flash_addr->data, ENV_SIZE) == flash_addr->crc;
 	crc2_ok =
 		crc32(0, flash_addr_new->data, ENV_SIZE) == flash_addr_new->crc;
+
 	if (crc1_ok && !crc2_ok) {
 		gd->env_addr	= addr1;
 		gd->env_valid	= 1;
@@ -125,9 +126,10 @@ int saveenv(void)
 	char	*res, *saved_data = NULL;
 	char	flag = OBSOLETE_FLAG, new_flag = ACTIVE_FLAG;
 	int	rc = 1;
-#if (CONFIG_ENV_SECT_SIZE > CONFIG_ENV_SIZE_FLASH)
-        ulong   up_data = 0;
-#endif /* (CONFIG_ENV_SECT_SIZE_FLASH > CONFIG_ENV_SIZE_FLASH) */
+#if CONFIG_ENV_SECT_SIZE > CONFIG_ENV_SIZE
+	ulong	up_data = 0;
+#endif
+
 	debug("Protect off %08lX ... %08lX\n", (ulong)flash_addr, end_addr);
 
 	if (flash_sect_protect(0, (ulong)flash_addr, end_addr))
@@ -145,10 +147,10 @@ int saveenv(void)
 		error("Cannot export environment: errno = %d\n", errno);
 		goto done;
 	}
-        env_new.crc     = crc32(0, env_new.data, ENV_SIZE);
+	env_new.crc	= crc32(0, env_new.data, ENV_SIZE);
 	env_new.flags	= new_flag;
 
-#if (CONFIG_ENV_SECT_SIZE > CONFIG_ENV_SIZE_FLASH)
+#if CONFIG_ENV_SECT_SIZE > CONFIG_ENV_SIZE
 	up_data = end_addr_new + 1 - ((long)flash_addr_new + CONFIG_ENV_SIZE);
 	debug("Data to save 0x%lX\n", up_data);
 	if (up_data) {
@@ -165,7 +167,7 @@ int saveenv(void)
 			(long)flash_addr_new + CONFIG_ENV_SIZE,
 			up_data, saved_data);
 	}
-#endif /* (CONFIG_ENV_SECT_SIZE_FLASH > CONFIG_ENV_SIZE_FLASH) */
+#endif
 	puts("Erasing Flash...");
 	debug(" %08lX ... %08lX ...", (ulong)flash_addr_new, end_addr_new);
 
@@ -186,16 +188,16 @@ int saveenv(void)
 	if (rc)
 		goto perror;
 
-#if (CONFIG_ENV_SECT_SIZE > CONFIG_ENV_SIZE_FLASH)
+#if CONFIG_ENV_SECT_SIZE > CONFIG_ENV_SIZE
 	if (up_data) { /* restore the rest of sector */
 		debug("Restoring the rest of data to 0x%lX len 0x%lX\n",
 			(long)flash_addr_new + CONFIG_ENV_SIZE, up_data);
 		if (flash_write(saved_data,
-                                (long)flash_addr_new + CONFIG_ENV_SIZE,
-                                up_data))
+				(long)flash_addr_new + CONFIG_ENV_SIZE,
+				up_data))
 			goto perror;
 	}
-#endif /* (CONFIG_ENV_SECT_SIZE_FLASH > CONFIG_ENV_SIZE_FLASH) */
+#endif
 	puts("done\n");
 
 	{
@@ -246,8 +248,9 @@ int saveenv(void)
 	ssize_t	len;
 	int	rc = 1;
 	char	*res, *saved_data = NULL;
-#if (CONFIG_ENV_SECT_SIZE > CONFIG_ENV_SIZE_FLASH)
+#if CONFIG_ENV_SECT_SIZE > CONFIG_ENV_SIZE
 	ulong	up_data = 0;
+
 	up_data = end_addr + 1 - ((long)flash_addr + CONFIG_ENV_SIZE);
 	debug("Data to save 0x%lx\n", up_data);
 	if (up_data) {
@@ -260,14 +263,17 @@ int saveenv(void)
 		memcpy(saved_data,
 			(void *)((long)flash_addr + CONFIG_ENV_SIZE), up_data);
 		debug("Data (start 0x%lx, len 0x%lx) saved at 0x%lx\n",
-			(ulong)flash_addr + CONFIG_ENV_SIZE, up_data, (ulong)saved_data);
+			(ulong)flash_addr + CONFIG_ENV_SIZE,
+			up_data,
+			(ulong)saved_data);
 	}
-#endif	/* (CONFIG_ENV_SECT_SIZE_FLASH > CONFIG_ENV_SIZE_FLASH) */
+#endif	/* CONFIG_ENV_SECT_SIZE */
+
 	debug("Protect off %08lX ... %08lX\n", (ulong)flash_addr, end_addr);
 
-	if (flash_sect_protect(0, (long)flash_addr, end_addr)) {
+	if (flash_sect_protect(0, (long)flash_addr, end_addr))
 		goto done;
-	}	
+
 	res = (char *)&env_new.data;
 	len = hexport_r(&env_htab, '\0', &res, ENV_SIZE, 0, NULL);
 	if (len < 0) {
@@ -275,27 +281,26 @@ int saveenv(void)
 		goto done;
 	}
 	env_new.crc = crc32(0, env_new.data, ENV_SIZE);
-	puts("Erasing Flash...");
 
+	puts("Erasing Flash...");
 	if (flash_sect_erase((long)flash_addr, end_addr))
 		goto done;
 
 	puts("Writing to Flash... ");
-
 	rc = flash_write((char *)&env_new, (long)flash_addr, CONFIG_ENV_SIZE);
-
 	if (rc != 0)
 		goto perror;
-#if (CONFIG_ENV_SECT_SIZE > CONFIG_ENV_SIZE_FLASH)
+
+#if CONFIG_ENV_SECT_SIZE > CONFIG_ENV_SIZE
 	if (up_data) {	/* restore the rest of sector */
 		debug("Restoring the rest of data to 0x%lx len 0x%lx\n",
 			(ulong)flash_addr + CONFIG_ENV_SIZE, up_data);
-
-		if (flash_write(saved_data, (long)flash_addr + CONFIG_ENV_SIZE, up_data)) {
-		        goto perror;
-		}
-    	}
-#endif /* (CONFIG_ENV_SECT_SIZE_FLASH > CONFIG_ENV_SIZE_FLASH) */
+		if (flash_write(saved_data,
+				(long)flash_addr + CONFIG_ENV_SIZE,
+				up_data))
+			goto perror;
+	}
+#endif
 	puts("done\n");
 	rc = 0;
 	goto done;
@@ -311,6 +316,7 @@ done:
 #endif /* CMD_SAVEENV */
 
 #endif /* CONFIG_ENV_ADDR_REDUND */
+
 void env_relocate_spec(void)
 {
 #ifdef CONFIG_ENV_ADDR_REDUND
diff --git a/include/dataflash.h b/include/dataflash.h
index e9f205d..94f86b3 100644
--- a/include/dataflash.h
+++ b/include/dataflash.h
@@ -136,7 +136,6 @@ struct dataflash_addr {
 /*-------------------------------------------------------------------------------------------------*/
 #define AT45DB161	0x2c
 #define AT45DB021	0x14
-#define AT45DB041	0x1c
 #define AT45DB081	0x24
 #define AT45DB321	0x34
 #define AT45DB642	0x3c
diff --git a/include/environment.h b/include/environment.h
index 6a21eb3..e8ab703 100644
--- a/include/environment.h
+++ b/include/environment.h
@@ -38,11 +38,11 @@
  */
 
 #if defined(CONFIG_ENV_IS_IN_FLASH)
-# ifndef        CONFIG_ENV_ADDR
-#  define       CONFIG_ENV_ADDR (CONFIG_SYS_FLASH_BASE + CONFIG_ENV_OFFSET)
+# ifndef	CONFIG_ENV_ADDR
+#  define	CONFIG_ENV_ADDR	(CONFIG_SYS_FLASH_BASE + CONFIG_ENV_OFFSET)
 # endif
-# ifndef        CONFIG_ENV_OFFSET
-#  define       CONFIG_ENV_OFFSET (CONFIG_ENV_ADDR - CONFIG_SYS_FLASH_BASE)
+# ifndef	CONFIG_ENV_OFFSET
+#  define	CONFIG_ENV_OFFSET (CONFIG_ENV_ADDR - CONFIG_SYS_FLASH_BASE)
 # endif
 # if !defined(CONFIG_ENV_ADDR_REDUND) && defined(CONFIG_ENV_OFFSET_REDUND)
 #  define	CONFIG_ENV_ADDR_REDUND	\
@@ -50,7 +50,7 @@
 # endif
 # if defined(CONFIG_ENV_SECT_SIZE) || defined(CONFIG_ENV_SIZE)
 #  ifndef	CONFIG_ENV_SECT_SIZE
-#   define CONFIG_ENV_SECT_SIZE CONFIG_ENV_SIZE
+#   define	CONFIG_ENV_SECT_SIZE	CONFIG_ENV_SIZE
 #  endif
 #  ifndef	CONFIG_ENV_SIZE
 #   define	CONFIG_ENV_SIZE	CONFIG_ENV_SECT_SIZE
@@ -58,7 +58,6 @@
 # else
 #  error "Both CONFIG_ENV_SECT_SIZE and CONFIG_ENV_SIZE undefined"
 # endif
-
 # if defined(CONFIG_ENV_ADDR_REDUND) && !defined(CONFIG_ENV_SIZE_REDUND)
 #  define CONFIG_ENV_SIZE_REDUND	CONFIG_ENV_SIZE
 # endif
@@ -139,7 +138,6 @@ extern unsigned long nand_env_oob_offset;
 extern char *env_name_spec;
 #endif
 
-#define ENV_SIZE_FLASH (CONFIG_ENV_SIZE_FLASH - ENV_HEADER_SIZE) /**< Martin bjærre <mab@rosetechnology.dk> */
 #define ENV_SIZE (CONFIG_ENV_SIZE - ENV_HEADER_SIZE)
 
 typedef struct environment_s {
diff --git a/include/flash.h b/include/flash.h
index b87837e..6d70bdd 100644
--- a/include/flash.h
+++ b/include/flash.h
@@ -260,7 +260,6 @@ extern flash_info_t *flash_get_info(ulong base);
 #define AMD_ID_GL128N_2	0x22212221	/* 2nd ID word for S29GL128N */
 #define AMD_ID_GL128N_3	0x22012201	/* 3rd ID word for S29GL128N */
 
-#define AMD_ID_GL512    0x227E2201      /* ID word for S29GL512P */
 
 #define AMD_ID_LV320B_2 0x221A221A	/* 2d ID word for AM29LV320MB at 0x38 */
 #define AMD_ID_LV320B_3 0x22002200	/* 3d ID word for AM29LV320MB at 0x3c */
@@ -474,8 +473,6 @@ extern flash_info_t *flash_get_info(ulong base);
 #define FLASH_MT28S4M16LC 0x00E1	/* Micron MT28S4M16LC			*/
 #define FLASH_S29GL064M 0x00F0		/* Spansion S29GL064M-R6		*/
 #define FLASH_S29GL128N 0x00F1		/* Spansion S29GL128N			*/
-#define FLASH_S29GL256P 0x00F2          /* Spansion S29GL256P                   */
-#define FLASH_S29GL512P 0x00F3          /* Spansion S29GL512P                   */
 
 #define FLASH_UNKNOWN	0xFFFF		/* unknown flash type			*/
 
diff --git a/tools/Makefile b/tools/Makefile
index fb6f88f..c31437e 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -147,6 +147,7 @@ endif
 ifeq ($(VENDOR),intercontrol)
 LOGO_BMP= logos/intercontrol.bmp
 endif
+
 # now $(obj) is defined
 HOSTSRCS += $(addprefix $(SRCTREE)/,$(EXT_OBJ_FILES-y:.o=.c))
 HOSTSRCS += $(addprefix $(SRCTREE)/tools/,$(OBJ_FILES-y:.o=.c))
-- 
1.8.3.1

